BE/src/Controllers/GetPackageControllers.ts

```diff
@@ -25,66 +25,58 @@ import {
 } from "ResponseTypes";
 import { NextFunction } from "express";
 import PackageModel from "../Schemas/Package";
+import { Buffer } from 'buffer';
 
 // /packages
 export const GetPackagesFromRegistryController = asyncHandler(
     async (req: GetPackagesRequest, res: GetPackagesResponse, next: NextFunction) => {
+        
         const requestedPackages = req.body;
-        //hover to see types
-        //your code here
-
-        //^^^^^^^^^^^^^^^^
-        //should return back to here a typed arr like this, this is metadata ab the package
-        const responseBody: GetPackagesResponseBody = [
-            { Version: "Some version", Name: "Some name", ID: "Some ID" },
-            { Version: "Some version", Name: "Some name", ID: "Some ID" },
-        ];
-        //something to signify too many packages were returned
-        const tooManyreturned = false;
+        const tooManyPackagesThreshold = 10; // Arbitrary Number, But Needed to Define One
+        const numberOfRequestedPackages = requestedPackages.length;
+
+        let responseBody: GetPackagesResponseBody[] = [];
         let responseMessage: GetPackagesInvalidResponseMessages;
-        if (!tooManyreturned) {
-            res.status(200).json(responseBody);
-        } else {
+
+        // Validate Input
+        if (numberOfRequestedPackages == 0)
+        {
+            responseMessage = "There is missing field(s) in the PackageQuery or it is formed improperly, or is invalid.";
+            res.status(404).send(responseMessage);
+        }
+        else if (numberOfRequestedPackages  >= tooManyPackagesThreshold) { // Check if too many packages
             responseMessage = "Too many packages returned.";
             res.status(413).send(responseMessage);
         }
-    }
-);
+
+        requestedPackages.forEach((requestedPackage) => {
+            // Checks if exists
+            PackageModel.findOne({ metadata: {name: requestedPackage.Name}})
+                .then((queriedPackage) => { // Have to do it this way since inside a forEach
```

Comment:
```ts
const queryConditions = requestedPackages.map((requestedPackageInfo) => ({
            "metadata.Name": requestedPackageInfo.Name,
            "metadata.Version": requestedPackageInfo.Version,
        }));
const results = await PackageModel.find({ $or: queryConditions });
```

 You will still have to transform to the proper return type. Please use this to eliminate the loop.